tests/unit/replication_persistance_helper.cpp

```diff
@@ -0,0 +1,199 @@
+                                                         .timeout = timeout,
+                                                         .replica_check_frequency = replica_check_frequency,
+                                                         .ssl = ssl};
+  }
+
+  bool CheckEquality(memgraph::storage::replication::ReplicaStatus &status,
+                     memgraph::storage::replication::ReplicaStatus &other_status) {
+    if (!CheckEqualityImpl(status, other_status)) {
+      ++number_of_tests_failed;
+      ExportFailingTest(status);
+      return false;
+    }
+
+    return true;
+  }
+
+  std::vector<std::string> names_ = {"name_01", "name_02"};
+  std::vector<std::string> ip_addresses_ = {"192.0.2.146", "168.212.226.204"};
+  std::vector<uint16_t> ports_ = {0, 1615, 5001};
+  std::vector<memgraph::storage::replication::ReplicationMode> sync_modes_ = {
+      memgraph::storage::replication::ReplicationMode::SYNC, memgraph::storage::replication::ReplicationMode::ASYNC};
+  std::vector<std::optional<double>> timeouts_ = {std::nullopt, 0, 125, 999.9};
+  std::vector<std::chrono::seconds> replica_check_frequencies_ = {std::chrono::seconds(0), std::chrono::seconds(99)};
+  std::vector<std::optional<memgraph::storage::replication::ReplicationClientConfig::SSL>> ssls_ = {
+      std::nullopt,
+      memgraph::storage::replication::ReplicationClientConfig::SSL{
+          .key_file = "MFsxCzAJBgNVBAYTAkZSMRMwEQYDVQQIDApTb21lLVN0YXRlMSEwHwYDVQQKDBhJ",
+          .cert_file = "20z0qSHpa3YNW6qSp"},
+      memgraph::storage::replication::ReplicationClientConfig::SSL{
+          .key_file = "+1ZyR4tXgi4+5MHGzhYCIVvHo4hKqYm+J+o5mwQInf1qoAHuO7CLD3WNa1sKcVUV",
+          .cert_file = "ZkChSRyc/Whvurx6o85D6qpzywo8xwNaLZHxTQPgcIA5su9ZIytv9LH2E+lSwwID"}};
+
+ private:
+  bool CheckEqualityImpl(memgraph::storage::replication::ReplicaStatus &status,
+                         memgraph::storage::replication::ReplicaStatus &other_status) {
+    if (status.name != other_status.name) {
+      return false;
+    }
+    if (status.ip_address != other_status.ip_address) {
+      return false;
+    }
+    if (status.port != other_status.port) {
+      return false;
+    }
+    if (status.sync_mode != other_status.sync_mode) {
+      return false;
+    }
+    if (status.timeout.has_value() != other_status.timeout.has_value()) {
+      return false;
+    }
+    if (status.timeout.has_value() && (*status.timeout != *other_status.timeout)) {
+      return false;
+    }
+    if (status.replica_check_frequency != other_status.replica_check_frequency) {
+      return false;
+    }
+    if (status.ssl.has_value() != other_status.ssl.has_value()) {
+      return false;
+    }
+    if (status.ssl.has_value()) {
+      if (status.ssl->key_file != other_status.ssl->key_file) {
+        return false;
+      }
+      if (status.ssl->cert_file != other_status.ssl->cert_file) {
+        return false;
+      }
+    }
+
+    return true;
+  }
```

Comment:
These are not needed, just default the operator of the struct `ReplicaStatus` like:

```
friend bool operator==(const ReplicaStatus&, const ReplicaStatus&) const = default;
```

See https://en.cppreference.com/w/cpp/language/default_comparisons